keps/sig-testing/3041-node-conformance-and-features/README.md

```diff
@@ -0,0 +1,782 @@
+  (marked as `[FeatureGate:Foo][Alpha]`) is supposed to be either `NodeConformance`
+  or `Feature`.
+
+### See also
+
+- https://github.com/kubernetes/kubernetes/issues/59001
+- https://docs.google.com/document/d/1BdNVUGtYO6NDx10x_fueRh_DLT-SVdlPC_SsXjYCHOE/edit?usp=sharing
+- https://github.com/kubernetes/community/blob/32a1c14d04ff78684d78b827ac7c49f70352d509/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests
+- https://groups.google.com/g/kubernetes-sig-testing/c/8NVCt9UoV_Q/m/Vqz3_TidAwAJ?utm_medium=email&utm_source=footer
+
+### User Stories (Optional)
+
+#### Story 1
+
+1. New feature gate is introduced as Alpha.
+1. Tests for this functionality are marked as `[FeatureGate:Foo][Alpha]`.
+1. New test grid tab is added to run these tests while enabling the feature gate explicitly.
+1. Feature gate is promoted to Beta and enabled by default.
+1. Tests for this functionality are marked as `[FeatureGate:Foo][Beta]`.
+1. Depending on whether test targets all environments or specific ones,
+   `NodeConformance` or `Feature:` labels are added. Note, the `Feature:` label
+   may (but not necessarily) match the `FeatureGate:` name.
+1. Test infra runs all these tests as part of Features or NodeConformance runs
+   to ensure that default installation of k8s has all the features working.
+1. Test infra runs all tests with every feature gate disabled, catching potential
+   GA features dependencies on the new functionality.
+1. Feature gate is promoted to GA. Feature gate is locked to the value and
+   cannot be disabled.
+1. Tests for this functionality are dropping the labels `[FeatureGate:Foo][Beta]`,
+   while keeping either `NodeConformance` or `Feature:` label.
+
+#### Story 2
+
+Decision making tree for the test labels:
+
+- Is test only works on Linux? Apply `[LinuxOnly]`
+- Is test validate functionality that is controlled by a Feature Gate? Apply `[FeatureGate:Foo][Alpha|Beta|Deprecated]`
+- Is test validate Core API that is enabled by default, GA, and works on any environment? Apply `[Conformance]`
+- Is test validate node-specific functionality that is enabled by default and works on "any" node? Apply `[NodeConformance]`
+- Is test only works when underlying container runtime has a specific feature enabled or specific node configuration is set? Apply `[Feature:Foo]` to describe this feature or configuration.
+- Can test only run on the "default" test infra node? If not, apply `[Environment:Foo]` to describe the specific environment that needs to be pre-configured.
+
+<!--
+
+STANDARD TEMPLATE AFTER THIS LINE, NO MEANINGFUL CONTENT
+
+-->
+
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
```

Comment:
breaking existing test workflows that key off the old labels?

Reply:
skew testing seems ripe for issues in particular

Reply:
we should note these.
the PRR section will also need some acknowledgement to merge I think